### ISSUE TITLE
truncate doesn't work as expected when the string contains `▪︎` character

### ISSUE THREAD

**Issue by batflarrow** [lodash/lodash#5761]
The output of the following piece of code is 10 instead of 5 i.e. the final length of the string after truncating is 10 instead of 5.
```
console.log(truncate('▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎',{length:5,omission:''}).length)
```

**Comment by hamzakhalidchouhdary** [comment]
Hello @batflarrow!,

After reviewing it, I've determined that the problem doesn't originate from lodash.truncate() itself but from the specific string provided. JavaScript interprets the special characters in the string as two characters each, leading to an inaccurate length calculation.

```
console.log(truncate('▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎',{length:5,omission:''})); // this will return 5 dots.
console.log('▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎'.length); // this will output 88 but these are 44 in total

```
To resolve this issue, I suggest examining and potentially adjusting the string to ensure that it reflects the desired character count accurately. If you have any further questions or encounter additional problems, please don't hesitate to reach out.

**Comment by batflarrow** [comment]
> Hello @batflarrow!,
> 
> After reviewing it, I've determined that the problem doesn't originate from lodash.truncate() itself but from the specific string provided. JavaScript interprets the special characters in the string as two characters each, leading to an inaccurate length calculation.
> 
> ```
> console.log(truncate('▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎',{length:5,omission:''})); // this will return 5 dots.
> console.log('▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎▪︎'.length); // this will output 88 but these are 44 in total
> ```
> 
> To resolve this issue, I suggest examining and potentially adjusting the string to ensure that it reflects the desired character count accurately. If you have any further questions or encounter additional problems, please don't hesitate to reach out.

Yeah, I realized javascript is treating this character with length 2 instead of 1, was wondering if we can somehow expect this to happen for other characters and change the logic in truncate and account for this fact. Haven't gone through the complete code, but I assume this should be doable. I can work on this and raise a pr... if we agree to fix this, and also agree that it has to be fixed in lodash. I'm not sure what logic will I have to implement to check if javascript is treating a single character with length > 1.

**Comment by hamzakhalidchouhdary** [comment]
@batflarrow! The dot in question appears to have been split into two elements, one of which is an empty string but not truly empty. This behavior doesn't seem to be related to lodash as the library is not designed to modify the string itself but to replace a substring within the string. The rest of the string should remain unchanged. You should explore an alternative workaround.

<img width="210" alt="Screenshot 2023-10-27 at 7 23 07 PM" src="https://github.com/lodash/lodash/assets/60621294/2cdefda6-46fc-4ce0-ac0b-544be3c71e9d">

**Comment by batflarrow** [comment]
> ▪︎

Yeah, I get that, but shouldn't the stringSize function account for this character as well?  if  [stringSize](https://github.com/lodash/lodash/blob/main/src/.internal/stringSize.ts) function that is being used [here](https://github.com/lodash/lodash/blob/aa18212085c52fc106d075319637b8729e0f179f/src/truncate.ts#L75) returns the length of the string correctly considering that this character has a length of 2, then the rest of the logic would work as expected, wouldn't it?
